src/request.rs

```diff
@@ -64,20 +64,31 @@ pub trait KvRequest: Sync + Send + 'static + Sized {
         stores
             .and_then(move |(key_data, store)| {
                 let request = self.make_rpc_request(key_data, &store);
+                let mut req = None;
+                let mut err = None;
+                match request {
+                    Ok(r) => req = Some(r),
+                    Err(e) => err = Some(e),
+                }
```

Comment:
So, I played around with this a lot, and I don't think it can be improved much - the borrowed/owned issue is (I think) impossible to work around. You can make the code a little bit more idiomatic by using `let (req, err) = match ...` here, rather than mutable variables, and matching on `req` on line 75, rather than using if/else,

Reply:
All right. Thanks.